### ISSUE TITLE
Gophish becomes unresponsive while delivering emails to recipients and gophish server becomes unavailable 

### ISSUE THREAD

**Issue by bhagoria** [gophish/gophish#1971]
What version of Gophish are you using?: Gophish v0.10.1

Brief description of the issue: Gophish becomes unresponsive when people start clicking on the link in numbers and emails still being delivered. During this time people who click on the phishing link observe page can not be displayed. Also, some slowness in email sending.

What are you expecting to see happen? : Gophish console remains up and running, no performance issues, emails being delivered without any issue, people who click on the phishing link should get on the landing page.

What are you seeing happen?
When I target 10K people in a batch (targetting more than 40K in 4 different batches) and launch the campaign and the emails start being delivered, I notice that the Gophish console becomes so slow, unable to refresh, people who click on the phishing link start getting page can not be displayed. And, Gophish service keeps running above 100%.

I would anticipate that the issue can be reproduced when people start clicking on the link in numbers and emails still being delivered.

I am using phishing domain running on https. 

Please provide any terminal output that may be relevant below:

I am sharing the Gophish error that can help us to look into it. 

2020/09/15 09:16:50 http: TLS handshake error from 117.X.X.X:32698: read tcp 81.X.X.X:443->117.X.X.X:32698: i/o timeout
2020/09/15 09:16:50 http2: server: error reading preface from client 157.47.X.X:51726: read tcp 81.x.x.x:443->157.X.45.1:51726: i/o timeout
2020/09/15 09:18:33 http: TLS handshake error from 34.93.X.X:38438: read tcp 81.x.x.x:443->34.X.X.38:38438: read: connection reset by peer
 Microsoft Outlook 16.0.5017; ms-office; MSOffice 16)\""
time="2020-09-15T09:22:49+02:00" level=warning msg="451 4.7.0 Timeout waiting for client input" code=451 email="\"Tonese\" <toniasdddd"
time="2020-09-15T09:23:17+02:00" level=warning msg="write tcp 81.x.x.x:33974->81.X.X.X:25: write: broken pipe" 

Please provide as many steps as you can to reproduce the problem:
 * Step 1 - Take 12 K targets in one batch, launch the campaign with a phishing link running on https with redirect enabled. While emails being delivered and people start clicking on the phishing link. You will notice so much traffic on the phishing server that would make Gophish unresponsive.
 16 GB ram is there on the phishing server, which does not seem like a root cause of this issue.

**Comment by glennzw** [comment]
I also noticed when doing some load testing on a 10k campaign that the UI becomes very unresponsive. I think one cause is that the API fetches all results (rather than using pagination); so the API call returns 10k artefacts, and then the datatable tries to load 10k artefacts in the browser. For me this killed Chrome, and the gophish process was very CPU hungry.

Actually, digging through the archives this is a known issue: https://github.com/gophish/gophish/issues/1534 https://github.com/gophish/gophish/issues/816 https://github.com/gophish/gophish/issues/29

I guess you'd need to run your test again and immediately close the UI to see if the backend problems persist or not. Mailhog is a nice utility to consume emails rather than sending them to the targets again.

**Comment by bhagoria** [comment]
Thanks, @glennzw I will try closing the UI as soon as I start the campaign but do you think it would also fix the other issue, which is the phishing link "page can not be displayed" due to load or not sure if this is the case here?

**Comment by glennzw** [comment]
I'm honestly not sure, but if the backend performs as expected when the UI is closed we know it's the pagination that's the root cause.

**Comment by bhagoria** [comment]
Update - I targetted 23K this time and immediately closed the browser after launching the campaign and waited until all the email delivered (continuously checked Gophish.error log file to see email delivery status), to my surprise I did not face any issue this time. You seem right. May be worth mentioning somewhere in the documentation to close the UI as soon as you launch the big campaigns until we have this issue sorted. Thanks for your help!
